### PR TITLE
Dsm ddp 8867 matrix questions table display

### PR DESCRIPTION
**Pull request by GiCharkviani**
https://broadinstitute.atlassian.net/browse/DDP-8867

After these changes, the matrix questions will be displayed in the following way:
Screenshot 2022-10-07 at 11 58 40

### CODE REVIEW

File: ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/components/matrix-answer-table.component.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  styleUrls: ['./matrix-answer.table.component.scss'],
+})
+export class MatrixAnswerTableComponent {
+  @Input() answers: MatrixAnswer[];
+  @Input() questionDefinition!: QuestionDefinition;
+
+  public isChecked(optionId: string, rowId: string): boolean {
+    const checkedAnswerIndex = this.answers
+      .findIndex(answer => answer.horizontalAnswer?.optionStableId === optionId && answer.verticalAnswer?.rowStableId === rowId);
+    return checkedAnswerIndex > -1;
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/components/matrix-answer-table.component.ts]: Hi Giorgi,
you can use `Array.prototype.some` for that purpose.
It returns boolean value.

File: ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/services/buildActivityData.service.ts [hunk 2]
```diff
@@ -30,11 +31,28 @@ export class buildActivityDataService {
...
+    private generateDefaultMatrixAnswers(question: QuestionDefinition): MatrixAnswer[] {
+      const matrixAnswer: MatrixAnswer[] = [];
+      question.rows.forEach(row => {
+        const mAnswer: MatrixAnswer = {
+          verticalAnswer: row,
+          horizontalAnswer: null
+        };
+        matrixAnswer.push(mAnswer);
+      });
+      return matrixAnswer;
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/services/buildActivityData.service.ts]: **_#optional_**

The method can be simplified a bit like:

```
 private generateDefaultMatrixAnswers(question: QuestionDefinition): MatrixAnswer[] {
     return question.rows.map(row => ({
      verticalAnswer: row,
      horizontalAnswer: null
    }));     
}
```

Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/services/buildActivityData.service.ts]: It is already so. I pushed change right after creating this PR. refresh and you will see too. Anyway, thanks

File: ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/services/buildingFactory.service.ts [hunk 3]
```diff
@@ -277,18 +278,18 @@ export class BuildingFactoryService {
...
+    return Object.entries(selectedAnswers['matrixSelected']).map(([questionId, answerId]): MatrixAnswer => ({
+        verticalAnswer: this.getMatrixVerticalAnswer(questionId, questDef['rows']),
+        horizontalAnswer: this.getMatrixHorizontalAnswer((answerId as string), questDef['options']),
       }));
   }
 
-  private getMatrixQuestion(questionId: string, rows: Array<Object>): string {
-    return rows.find(question => question['rowStableId'] === questionId)['rowText'];
+  private getMatrixVerticalAnswer(questionId: string, rows: Row[]): Row {
+    return rows.find(question => question['rowStableId'] === questionId);
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/services/buildingFactory.service.ts]: `question.rowStableId`  is better for finding some type errors then `question['rowStableId']`.
The same below.

File: ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/components/matrix-answer-table.component.ts (hunk 1 above)
Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/components/matrix-answer-table.component.ts]: Hi Alex. Thanks, good idea. Done!

File: ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/services/buildingFactory.service.ts (hunk 3 above)
Comment (GiCharkviani) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/services/buildingFactory.service.ts]: Actually, it is already changed :D the same happened, as I mentioned above in comment. Thanks.

**Review by alextanlan — APPROVED** [review]

**Pull request merged**